includes/assets/stories-widget.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function ($) {
+  /**
+   * Store field states in local variable.
+   *
+   * @type {Object}
+   */
+  var fieldState = window.webStoriesData || {};
+
+  /**
+   * Closure for making inputs reactive
+   * on view type selection change.
+   *
+   * @param {Object}  target Event object.
+   * @param {boolean} reset Flag to reset fields value.
+   */
+  var react = function (target, reset = false) {
+    var currentView = target.value;
+    var widget = target.closest('.widget');
+    var state = fieldState.fields[currentView];
+
+    for (var [key, value] of Object.entries(state)) {
+      var field = widget.querySelector('.' + key + '.stories-widget-field');
+      var fieldWrapper = widget.querySelector('.' + key + '_wrapper');
+      if (field && fieldWrapper && 'checkbox' === field.getAttribute('type')) {
+        if (reset) {
+          field.checked = false;
+        }
+        /**
+         * If this is readonly field.
+         * Assign the value automatically and hide it afterward.
+         */
+        if (value.readonly) {
+          field.checked = value.show;
+        }
+        fieldWrapper.style.display = value.readonly ? 'none' : 'block';
+      }
+    }
+  };
+
+  /**
+   * Returns the HTML Collection of view type selectors.
+   *
+   * @return {HTMLCollection} Array of elements.
+   */
+  var viewSelectors = function () {
+    return document.getElementsByClassName('view-type stories-widget-field');
+  };
+
+  /**
+   * Bind event on view type selection change.
+   */
+  var bindEvent = function () {
+    var dropdowns = viewSelectors();
+
+    if (dropdowns.length) {
+      for (var i = 0; i < dropdowns.length; i++) {
+        dropdowns[i].onchange = function (event) {
+          react(event.target, false);
+        };
+      }
+    }
+  };
+
+  /**
+   * Fire event manually once.
+   */
+  var fireEvent = function () {
+    var dropdowns = viewSelectors();
+    var evt = document.createEvent('HTMLEvents');
+    evt.initEvent('change', false, true);
+    for (var l = 0; l < dropdowns.length; l++) {
+      dropdowns[l].dispatchEvent(evt);
+    }
+  };
```

Comment:
Why do we need to do this?

Reply:
When we load widget page in admin for the first time, we have to manually trigger the `onChange` event for `Layout` dropdown, so that only relevant controls (checkboxes) would be visible. 

In `bindEvent` function, we are binding `onChange` event to Layout dropdown. 

In `fireEvent` function we are dispatching onChange event on initial load of the page.